### ISSUE TITLE
07-naive-bayes.ipynb

### ISSUE THREAD

**Issue by mezcolantriz** [4GeeksAcademy/adamcn10-intro-ml#5]
@AdamCN10 

¡Estructura y organización de 10! 

- El notebook está muy bien organizado con secciones claras
- Mucha documentación markdown y bien estructurada
- Flujo lógico desde la carga de datos hasta el guardado de modelos

- Comprendes que Naive Bayes es apropiado para análisis de sentimientos
- Identificas correctamente que MultinomialNB es mejor para datos de texto
- Buena justificación de por qué no optimizas GaussianNB

- Uso adecuado de CountVectorizer con stop_words="english"
- Implementación correcta de los tres tipos de Naive Bayes
- GridSearchCV implementado perfect para optimización

---
# Errores

1. Falta *stratify* en train_test_split 

<img width="1160" height="313" alt="Image" src="https://github.com/user-attachments/assets/864f3ca4-b45d-4415-a913-4a56579f856b" />

```py X_train, X_test, y_train, y_test = train_test_split(X, y,                                                     test_size=0.2,                                                     random_state=25)# DEBERÍA SER:X_train, X_test, y_train, y_test = train_test_split(X, y,                                                     test_size=0.2,                                                     random_state=25,                                                    stratify=y)
```
---

2. *Métricas*:

- Solo usas accuracy como métrica
- No incluyes precision, recall, F1-score
- Falta matriz de confusión o classification_report, intenta ponerlas siempre que puedas.

---

3. Falta de Análisis de Distribución de Clases.

- No verificas el balance de las clases
- No analizas si el dataset está balanceado
